inference-engine/src/mkldnn_plugin/nodes/mkldnn_tile_node.cpp

```diff
@@ -3,74 +3,91 @@
+    if ((tile && getParentEdges().size() != 2) || (!tile && getParentEdges().size() != 1))
         THROW_IE_EXCEPTION << "Incorrect number of input edges for layer " << getName();
     if (!getChildEdges().size())
         THROW_IE_EXCEPTION << "Incorrect number of output edges for layer " << getName();
+    if (getParentEdgeAt(0)->getDims().ndims() > getChildEdgeAt(0)->getDims().ndims())
+        THROW_IE_EXCEPTION << "Incorrect shapes for layer " << getName() << ". Input shape cannot be more than output shape";
+
+    if (tile) {
+        auto repeatsBlob = dynamic_cast<TBlob<int> *>(getParentEdgeAt(1)->getParent()->getCnnLayer()->blobs["custom"].get());
+        if (!repeatsBlob) {
+            THROW_IE_EXCEPTION << "Cannot get repeatsBlob for layer " << getName();
+        }
+        auto repeatsData = repeatsBlob->buffer().as<int *>();
+
+        for (int i = 0; i < getParentEdgeAt(1)->getDims().ToSizeVector()[0]; i++) {
+            repeats.push_back(repeatsData[i]);
+        }
+        while (repeats.size() < getChildEdgeAt(0)->getDims().ndims()) {
+            repeats.insert(repeats.begin(), 1);
+        }
 
-    axis = tileLayer->axis;
-    tiles = tileLayer->tiles;
+        optimizedCase = isOptimizedCase(repeats);
+    } else {
+        axis = tileLayer->axis;
+        tiles = tileLayer->tiles;
+    }
 }
 
 void MKLDNNTileNode::initSupportedPrimitiveDescriptors() {
     if (!supportedPrimitiveDescriptors.empty())
         return;
 
-    InferenceEngine::Precision precision = getCnnLayer()->insData[0].lock()->getPrecision();
-    if (precision != InferenceEngine::Precision::FP32)
-        precision = InferenceEngine::Precision::FP32;
-    auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
-    precision = getCnnLayer()->outData[0]->getPrecision();
-    if (precision != InferenceEngine::Precision::FP32)
-        precision = InferenceEngine::Precision::FP32;
-    auto outputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
-
-    auto& inDims = getParentEdgeAt(0)->getDims();
-    memory::format fmt = memory::format::any;
-    if (inDims.ndims() == 1) {
-        fmt = memory::format::x;
-    } else if (inDims.ndims() == 2) {
-        fmt = memory::format::nc;
-    } else if (inDims.ndims() == 3) {
-        fmt = memory::format::tnc;
-    } else if (inDims.ndims() == 4) {
-        fmt = memory::format::nchw;
-    } else if (inDims.ndims() == 5) {
-        fmt = memory::format::ncdhw;
-    }
-    if (fmt == memory::format::any) {
-        THROW_IE_EXCEPTION << "Tile " << getName() << " supports only 2D, 4D and 5D dimensions!";
-    }
+    if (tile) {
+        supportedPrimitiveDescriptors = getSupportedConfigs(this, optimizedCase, repeats);
+    } else {
+        InferenceEngine::Precision precision = getCnnLayer()->insData[0].lock()->getPrecision();
+        if (precision != InferenceEngine::Precision::FP32)
+            precision = InferenceEngine::Precision::FP32;
```

Comment:
Why we cannot set FP32 directly for input and output without retrieving and check precision from Data?

Reply:
You're right, fixed